apps/news/static/layout.css

```diff
@@ -1,82 +1,111 @@
-body  { font-family:Verdana, Geneva, sans-serif; font-size:10pt; }
+body  {
+  font-family:Verdana, Geneva, sans-serif;
+  font-size:  calc( (1rem) * 10 / 12 );
+}
 
 a:link { text-decoration:none; color:#828282; }
 a:visited { text-decoration:none; color:#000000; }
 
-.subtext { font-size:85%; padding-left: 15px; left:1.4em; margin-bottom: 0.5em; color:#828282 }
+.subtext { font-size:85%; padding-left: 1em; left:1.4em; margin-bottom: 0.5em; color:#828282 }
 
 form {
 	padding-top:1em;
 	padding-bottom:1em;
-	padding-left:2em;
+  padding-left: 1em;
+  padding-right: 1em;
+
+  @media (min-width: 640px) {
+    padding-left:2em;
+    padding-right:2em;
+  }
+}
+
+input[type='text'], input[type='number'], textarea {
+  max-width: 100%;
+}
+
+td > input[type='text'], td > input[type='number'], td > textarea {
+  max-width: 80%;
+}
+
+form[method=post] tr td {
+  max-width: 70%;
 }
 
 .layout {
-	width:85%;
-	min-width:85%;
-	margin:0px auto;
-	display:block;
+  background-color: #ddd;
+  margin: 0 auto;
+  max-width: 1024px;
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
 }
 
 .page-header,.page-footer {
-    display: block;
-    padding-top: 2px;
-    padding-bottom: 2px;
-    height: 18px;
+    min-height: 1.5em;
+    width: 100%;
     margin: 0 auto;
-    padding-left: 2px;
+    flex-basis: fit-content;
 }
 
 .page-header a:visited { color:#000000; }
 
 #logo {
-	margin-right:4px;
+	margin-right: 0.25em;
 }
 
 #page-title {
 	position:relative;
-	bottom:4px;
+	bottom: 0.25em;
 	font-weight:bold;
 }
 
 li.no-number { list-style-type: none; }
 
-#navmain {
-	width:auto;
-	padding-left:1em;
-	position:relative;
-	height:10px;
-	border:0;
-	margin-top:0;
-	margin-bottom:0;
-	bottom:4px;
+.page-header {
+  display: flex;
+}
+
+.page-header > * {
+  padding: 0.125em 0.25em;
+  line-height: 1.5em;
 }
 
-#navright {
-	float:right;
-	margin-right:0.5em;
-	padding-left:1em;
+.page-footer > * {
+  display: block;
+  line-height: 1.5em;
+  padding: 0.125em 0.25em;
+}
+
+#navmain {
+  flex-grow: 1;
```

Comment:
When you have sibling elements inside a `flex`, the value `flex-grow` is a great way to have a specific element say "if there is extra space, I should fill it."